src/coreclr/debug/createdump/crashinfo.cpp

```diff
@@ -188,9 +210,13 @@ CrashInfo::GatherCrashInfo(MINIDUMP_TYPE minidumpType)
             for (const MemoryRegion& region : m_otherMappings)
             {
                 uint32_t permissions = region.Permissions();
+#ifdef __APPLE__
+                if (permissions == (PF_R | PF_W))
+#else
                 if (permissions == (PF_R | PF_W) || permissions == (PF_R | PF_W | PF_X))
+#endif
                 {
-                    InsertMemoryBackedRegion(region);
+                    InsertMemoryRegion(region);
```

Comment:
Why are we excluding RWX regions on macOS? This should not be all too uncommon?

Reply:
I'll remove that change. I was thinking that were a lot of them, but there isn't.

Reply:
On M1 it does end up being a lot of 16K pages (10MB in my simple test case) so I'm not going to change this back.

Reply:
On m1 R2R code pages are mapped as RWX. Does `MiniDumpWithPrivateReadWriteMemory` imply no module code pages? (I hadn't realized that it was this part of the code).

Reply:
Yes, it means just rw heap pages.  Most code pages are not part of a heap dump with a couple of exceptions.